luigi/scheduler.py

```diff
@@ -737,12 +740,29 @@ def dump(self):
         if self._config.batch_emails:
             self._email_batcher.send_email()
 
+    def _prune_short_lived_stakeholders(self):
+        if self._config.short_lived_stakeholders:
+            useful_stakeholders = set()
+            all_stakeholders = set()
+            for task in self._state.get_active_tasks():
+                all_stakeholders.update(task.stakeholders)
+                # Tasks that are not done or disabled are useful
+                if task.status != DONE and task.status != DISABLED:
```

Comment:
this still can cause a problem if we have tasks like
LarryTask depends on CaseyTask and CaseyTask finishes but LarryTask fails.  As soon as LarryTask fails, the timer for removing CaseyTask will begin.  If we do not reenable LarryTask in time, CaseyTask will disappear.  We can rerun manually, and maybe that's just something we need to know

Reply:
Hmm, I think I understand -- and, what's more, even if LarryTask is reenabled and goes back to `RUNNING`, it won't re-add the stakeholder to CaseyTask, right? So once LarryTask fails there's no stopping CaseyTask from being removed

Reply:
However, I suppose this can only happen for the HEAD of the DAG, since any failure anywhere else in the dag would result in `UPSTREAM_DISABLED` tasks, and those we don't count as DONE enough to remove stakeholders from

Reply:
yes on both comments @brandonAIQ .  from my observations upstream_disabled is actually 'pending' within the luigi scheduler

Reply:
Hmmmm, okay that puts me at ease a bit. We still need to make sure we have a plan when this happens (I guess just `force_run_task`) but it's better since it'll be rare.